Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemGroupsBusinessService.cs

```diff
@@ -76,6 +79,33 @@ await sourceContestProblemGroups
             return ServiceResult.Success;
         }
 
+        public async Task ReevaluateProblemsAndProblemGroupsByOrderBy(int contestId)
+        {
+            var orderByIndex = 0;
+
+            this.problemGroupsData.GetAllNonDeletedByContest(contestId)
```

Comment:
You can just get the ProblemGroups with the Problems with `.Include()` instead of making additional query for the Problems